library/fluid/fluid_extensions/src/main/java/org/quiltmc/qsl/fluid/api/QuiltFlowableFluidExtensions.java

```diff
@@ -0,0 +1,532 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.fluid.api;
+
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.entity.effect.StatusEffectUtil;
+import net.minecraft.entity.effect.StatusEffects;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.FishingBobberEntity;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.LavaFluid;
+import net.minecraft.fluid.WaterFluid;
+import net.minecraft.loot.LootTables;
+import net.minecraft.particle.ParticleEffect;
+import net.minecraft.particle.ParticleTypes;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.tag.FluidTags;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.random.RandomGenerator;
+import net.minecraft.world.World;
+import net.minecraft.world.event.GameEvent;
+import org.quiltmc.qsl.base.api.util.InjectedInterface;
+
+import javax.annotation.Nullable;
+
+@InjectedInterface({WaterFluid.class, LavaFluid.class})
+public interface QuiltFlowableFluidExtensions {
+
+	float WATER_VISCOSITY = 0.8f;
+	float LAVA_VISCOSITY = 0.5f;
+	float WATER_PUSH_STRENGTH = 0.014f;
+	float LAVA_PUSH_STRENGTH_OVERWORLD = 7 / 3000f;
+	float LAVA_PUSH_STRENGTH_ULTRAWARM = 0.007f;
+	float WATER_DENSITY = 1000f;
+	float LAVA_DENSITY = 3100f;
+	float WATER_TEMPERATURE = 300f;
+	float LAVA_TEMPERATURE = 1500f;
+	float WATER_FOG_START = -8.0f;
+	float FULL_FALL_DAMAGE_REDUCTION = 0f;
+	float HALF_FALL_DAMAGE_REDUCTION = 0.5f;
+	float NO_FALL_DAMAGE_REDUCTION = 0f;
+	int WATER_FOG_COLOR = -1;
+
+	int LAVA_FOG_COLOR = 0x991900;
+
+	Identifier WATER_FISHING_LOOT_TABLE = LootTables.FISHING_GAMEPLAY;
+
+	int DROWNING_THRESHOLD = -20;
```

Comment:
Would it be possible to link these to constants within minecrafts code? or all of these values inlined with no public/private static final field?